glib/src/closure.rs

```diff
@@ -393,6 +390,63 @@ impl<T: for<'a> FromValue<'a> + StaticType + 'static> TryFromClosureReturnValue
 unsafe impl Send for Closure {}
 unsafe impl Sync for Closure {}
 
+pub trait SignalClosure<A> {
+    type Output: ToClosureReturnValue;
+
+    fn call(&self, args: &[Value]) -> Option<Value>;
+    fn parameter_types() -> Vec<Type>;
+}
+
+macro_rules! signal_closure_impl ({ $($param:ident)* } => {
+    impl<Func: Fn($($param),*) -> Res, Res: ToClosureReturnValue, $($param: StaticType + for<'a> FromValue<'a>,)*> SignalClosure<($($param,)*)> for Func
```

Comment:
Because of the `for<'a>` it is currently required that each parameter is `'static`, e.g. `&str` can't be used.

Fixing this requires adding a lifetime parameter to the trait and the parameter of the `call()` function, but I couldn't figure out how to make everything match up then.

Reply:
Also asked on users.rust-lang.org: https://users.rust-lang.org/t/problems-matching-up-lifetimes-between-various-traits-and-closure-parameters/71994

Reply:
There's some promising comments in that thread. I think maybe the same approach would also work for https://github.com/gtk-rs/gtk-rs-core/issues/382

Reply:
Yeah, I'll have to find some time to try this out :)